libraries/AP_Generator/AP_Generator_Cortex.cpp

```diff
@@ -0,0 +1,352 @@
+    }
+
+    // Warning flags
+    if (warning.generatorTempLimit) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_OVERTEMP_WARNING;
+    }
+
+    if (warning.temperature) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_ELECTRONICS_OVERTEMP_WARNING;
+    }
+
+    if (warning.enginePowerLimit) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_MAXPOWER;
+    }
+    
+    // Error flags
+    if (error.generator) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_ELECTRONICS_FAULT;
+    }
+
+    if (error.avionicsRail ||
+        error.payloadRail ||
+        error.servoRail ||
+        error.batteryChargerRail) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_POWER_RAIL_FAULT;
+    }
+
+    mavlink_msg_generator_status_send(
+        channel.get_chan(),
+        status_flags,
+        (uint16_t) abs(telemetry.generator.rpm),
+        batteryCurrent(),
+        loadCurrent(),
+        generatorPower(),
+        generatorVoltage(),
+        rectifierTemperature(),
+        std::numeric_limits<double>::quiet_NaN(),
+        telemetry.generator.temperature,
+        telemetry.controller.runTime,
+        0  // time until maintenance (not supported)
+    );
+#endif // HAL_PICCOLO_CAN_ENABLE
+}
+
+
+bool AP_Generator_Cortex::is_connected(void) const
+{
+    const uint32_t now = AP_HAL::millis();
+
+    return (now - last_reading_ms) < 1000;
+}
+
+
+bool AP_Generator_Cortex::is_inhibited(void) const
+{
+#if HAL_PICCOLO_CAN_ENABLE
+    return telemetry.status.status.inhibited;
+#else
+    // Fallback in the case that PiccoloCAN support is not enabled
+    return false;
+#endif // HAL_PICCOLO_CAN_ENABLE
+}
+
+
+bool AP_Generator_Cortex::is_ready(void) const
+{
+#if HAL_PICCOLO_CAN_ENABLE
+    return is_connected() && telemetry.status.status.readyToRun;
+#else
+    // Fallback in the case that PiccoloCAN support is not enabled
+    return false;
+#endif // HAL_PICCOLO_CAN_ENABLE
+}
+
+
+bool AP_Generator_Cortex::healthy() const
+{
+    if (!is_connected()) {
+        return false;
+    }
+
+    return true;
+}
+
+#if HAL_PICCOLO_CAN_ENABLE
+int16_t AP_Generator_Cortex::rectifierTemperature(void) const
+{
+    const Cortex_TelemetryController_t &controller = telemetry.controller;
+
+    // Return the maximum of the two internal temperature values
+    return controller.rectifierTemperature > controller.regulatorTemperature ?
+        controller.rectifierTemperature : controller.regulatorTemperature;
+}
+#endif // HAL_PICCOLO_CAN_ENABLE
+
+
+bool AP_Generator_Cortex::send_message(AP_HAL::CANFrame &frame)
+{
+#if HAL_PICCOLO_CAN_ENABLE
+    if (_can_iface == nullptr) {
+        return false;
+    }
+
+    if (!is_connected()) {
+        return false;
+    }
+
+    // Encode stored device type and node ID
+    frame.id |= _can_device_id;
+    frame.id |= (_can_device_type << 8) & 0xFF00;
+
+    return _can_iface->write_frame(frame, 1000);
+#else
+    // PiccoloCAN interface not supported
+    return false;
+#endif // HAL_PICCOLO_CAN_ENABLE
```

Comment:
```suggestion
```

... and everywhere else.  Just assume in this file that `HAL_PICCOLO_CAN_ENABLE` is set